### DISCUSSION TITLE
How do I make the results of the polygon to be data overlay

### DISCUSSION THREAD

**Discussion by Fakhrynm** [opengeos/leafmap#474] [Q&A] [answered]
I made a WebGIS for my final project. But I need help when I want to retrieve data based on the results of the polygons I created on the map. I want to retrieve existing data in the database based on polygon overlays. I need clarification about how to call Polygon if the data is in self.draw_feature.
<img width="492" alt="image" src="https://github.com/opengeos/leafmap/assets/93865703/1152a7fa-d3c4-4dcb-a62e-3831d420c6e8">

**Comment by giswqs** [comment, marked as answer]
`m.user_roi`

**Reply by Fakhrynm** [reply]
Thank you, sir. I will ask you again later.

**Reply by Fakhrynm** [reply]
Hello sir. I have tried call back using 
m.user_roi 

but the result m.user_roi does not update after I draw polygon.

Here my code:

import streamlit as st
import psycopg2
import leafmap
import geopandas as gpd

# Connect to the database
# def app():
conn = psycopg2.connect(
    database="Testing",
    user="postgres",
    password="admin",
    host="localhost",
    port="5432"
)
cursor = conn.cursor()

# Retrieve data from the database
cursor = conn.cursor()
query = "SELECT latitude, longitude FROM spasial"
cursor.execute(query)
data = cursor.fetchall()

# Create a Leafmap

m = leafmap.Map()
# m = leafmap.Map(locate_control=True)
m.add_basemap("ROADMAP")

# Get the drawn polygon from the Leafmap and convert it to a GeoDataFrame:
polygon = m.user_roi
print(polygon)
gdf_polygon = gpd.GeoDataFrame([polygon], columns=['geometry'])

# # Iterate over the data and add markers
# for record in data:
#     latitude, longitude = record[0], record[1]
#     m.add_marker(location=[latitude, longitude])

# Convert the drawn polygon to Well-Known Text (WKT) format
wkt_polygon = gdf_polygon['geometry'].to_wkt()[0]

# Perform a spatial intersection query using the drawn polygon and the database data:
intersection_query = "SELECT latitude, longitude FROM spasial WHERE ST_Intersects(ST_GeomFromText(%s), geometry)"
cursor.execute(intersection_query, [wkt_polygon])
intersected_data = cursor.fetchall()


#Function for display masrker
def display_data():
    
    if st.button(label="Display Data"):
        # Display the intersected data on the Leafmap:
        for row in intersected_data:
            lat, lon = row
            marker = leafmap.Marker(location=(lat, lon))
            m.add_layer(marker)
    
    polygon = m.user_roi  # Update the `polygon` variable with the drawn polygon
    print(polygon)

# Call the download_file function
display_data()
# Display the map

m.to_streamlit(height=700)

Here is a picture after I draw a rectangle
<img width="730" alt="image" src="https://github.com/opengeos/leafmap/assets/93865703/229c9484-5ef7-4494-8d77-293144e9c97f">

Here is a picture after I click my function to show the data and the value of m.user_roi still none
<img width="436" alt="image" src="https://github.com/opengeos/leafmap/assets/93865703/073908ce-0b57-48a6-8a86-3876ede3169c">

The rectangle also will be lost.
<img width="744" alt="image" src="https://github.com/opengeos/leafmap/assets/93865703/54856455-fe0b-4313-8f9c-9523f7ba1efe">
